### ISSUE TITLE
[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.

### ISSUE THREAD

**Issue by jvbianchi** [angular/material2#4221]
#### Bug, feature request, or proposal:
In chrome 58, warnings started appearing on console:

```
zone.min.js:1 [Violation] 'readystatechange' handler took 190ms
zone.min.js:1 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.
zone.min.js:1 [Violation] 'load' handler took 517ms
```

#### What is the expected behavior?
Don't get this warning on console:
![screen shot 2017-04-22 at 20 02 47](https://cloud.githubusercontent.com/assets/13983663/25308965/ba8ea0ba-2796-11e7-99e0-f307c78f7deb.png)

#### What are the steps to reproduce?
Just running the template plunker you get this console warning.
**Console must be in the verbose level**

Plunker: https://goo.gl/DlHd6U

#### Which versions of Angular, Material, OS, browsers are affected?
Chrome 58.0.3029.81
Mac OSX 10.12.4 (Sierra)

**Comment by jackvial** [comment]
Looks like it's related to the new event listener options. More on it [here](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md) 

There is no a new options object that can be passed to the addEventListener. The warning is saying the passive event listeners on the page should be passed the {passive: true} option to improve scroll performance. `document.addEventListener('touchstart', handler, {passive: true});`.  

These warnings are only performance recommendation so it's not really something to get too worried about, although performances improvements are always good.

**Comment by tarlepp** [comment]
Using https://github.com/zzarcon/default-passive-events will remove some of those violation warnings, but there is still some left.

**Comment by mackelito** [comment]
Is this really a matter of material? or is this with zone.js? Would be great if someone from the material team could verify :)

**Comment by acronHou** [comment]
what about "bind"?

**Comment by Zwilla** [comment]
I can not understand why this is closed, the problem always exist and only this:
https://github.com/zzarcon/default-passive-events
solved my problem, the above problem
thx to @tarlepp

**Comment by Dunos** [comment]
Yup, can confirm that this is still happening

**Comment by FredvanRijswijk** [comment]
Yes still getting this warning:

zone.js:1666 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.

**Comment by mackelito** [comment]
@andrewseguin are there any recommended solutions for this issue?

**Comment by micahwood** [comment]
This seems to have gotten worse with beta 12. Displaying a dialog used to appear instantly, now it takes close to 2 seconds for the dialog to appear.
`[Violation] 'click' handler took 1690ms`

The following message is logged about 1200 times from the time I click the button to display a dialog and it appearing.
`[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.`

**Comment by micahwood** [comment]
Here is a plunker showing the issue with slowing down substantially with large data sets. I added a button to display the autocomplete to make it easier to see the slow down. Also if you open the dev tools and set logging to verbose you can see how many event listeners are being created. It looks like it creates two listeners for each item that is added. I will create a new issue for this.
https://plnkr.co/edit/QhlHy7FrlvQbDeRfncCo?p=preview

**Comment by crisbeto** [comment]
FWIW, those `touchstart` handlers were added to accommodate the ripple effect on buttons. They have since been removed because they were problematic.

**Comment by tarlepp** [comment]
this is still happening, should we create a new issue about this, because this one is "closed" when it really isn't resolved at anyway ?

**Comment by d3viant0ne** [comment]
@tarlepp - This should have been resolved with #7557 ( doesn't happen on master ) so it should just be a matter of waiting on the next tag or running on a nightly build if it's absolutely critical.

**Comment by debslord** [comment]
Thanks @d3viant0ne do you know when this is likely to be released?  

I'm getting a lot of [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' events appearing in the console and my forms are really slow to load.... It reports back that 
[Violation] 'click' handler took 17807ms

Keeping my fingers crossed that #7557 resolves these issues!

**Comment by d3viant0ne** [comment]
I'm running on `"@angular/material": "git+https://github.com/angular/material2-builds#2.0.0-beta.12-62b149f",` & the associated CDK & the problem no longer exists.

That build is stable & works with Angular 5.x, at least within the scope of what I am doing. material2-build are static if you target the hash, really no different than a published build in the strictest sense.

**Comment by debslord** [comment]
Thanks again @d3viant0ne 

Upgraded my project to Angular 5 and the build you're on - weirdly still an issue for me.  

This occurs 8000+ times - [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event.  
[Violation] 'click' handler took 18268ms
[Violation] 'load' handler took 477ms

As a result the form takes around 20 seconds to load - same as @micahwood plunker as he said the larger the dataset the more event listeners

**Comment by d3viant0ne** [comment]
@debslord - I'll take another look with the filters off. 90% sure the `touchstart` stuff is no longer an issue ( on my side at least )

**Comment by debslord** [comment]
Thank you! I will see about implementing default-passive-events I might have to result to that fix instead

**Comment by debslord** [comment]
I think my issues are probably more around the autocomplete issues mentioned in #5113 and #7749 
So it might be slightly different to the issues you were getting @d3viant0ne 

Implementing default-passive-events doesn't work either as there seems to be an issue around click events already having this set by default and so it throws an error instead and although the toughstart issues aren't displayed in the console the click still takes much too long.  I'm hoping the other two issues will resolve this.  Thanks for you help.

**Comment by lexyfeito** [comment]
Please when is this going to be fixed, this is slowing down my app's performance like crazy. Thank you

**Comment by vegarda** [comment]
This is back in 5.0.0-rc.3(?)

**Comment by olaf89** [comment]
I have same issues with 5.0.0-rc.3

**Comment by crisbeto** [comment]
It should be fixed by https://github.com/angular/material2/pull/8719.

**Comment by lexyfeito** [comment]
@crisbeto which release will this be in?

**Comment by paqogomez** [comment]
Could someone please explain how they are implementing @zzarcon's passive event patch in angular?  I'm missing something very basic I think.

**Comment by gyx8899** [comment]
I find one solution to solve this warning.  (It works fine for me)

[Added-non-passive-event-listener-to-a-scroll-blocking-touchstart-event-Consider-marking-event-handler-as-passive-to-make-the-page-more-responsive-by-visionfortech](https://visionfortech.blogspot.jp/2017/07/Added-non-passive-event-listener-to-a-scroll-blocking-touchstart-event-Consider-marking-event-handler-as-passive-to-make-the-page-more-responsive-by-visionfortech.html)

`jQuery.event.special.touchstart = {
  setup: function( _, ns, handle ){
   if ( ns.includes("noPreventDefault") ) {
     this.addEventListener("touchstart", handle, { passive: false });
   } else {
     this.addEventListener("touchstart", handle, { passive: true });
   }
  }
};`

**Comment by fertandil87** [comment]
it's fixed in 5.0.0-rc.2, and appear again in  ">= 5.0.0-rc.3"

**Comment by surbhi290391** [comment]
i am facing this same issue in recaptcha_en.js s can anyone tell me ow can i enable passive=true for recaptcha

**Comment by phil123456** [comment]
sept 2018 and still hapenning
 angular: ^5.2.0
 ng CLI: ~1.7.4
 tslint: ~5.9.1
 typescript: ~2.5.3
 webpack: ~3.11.0

**Comment by crisbeto** [comment]
@phil123456 which Material components are you using on the page that is logging out the warning?

**Comment by brecko** [comment]
I was having the same issue. It turned out to be the Chrome Extension 'Logitech Smooth Scrolling'. When I turned it off, I no longer saw the violations in my Angular 6 Material app.

**Comment by krokofant** [comment]
The *default-passive-events* workaround is horrible, it messes up the available functionality of existing or future eventhandlers that want to `preventDefault()` an event.

**Comment by mysexyDate24com** [comment]
Any updates on this issue?
